components/camera/rtsp/rtsp_test.go

```diff
@@ -0,0 +1,54 @@
+package rtsp
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/bhaney/rtsp-simple-server/server"
+	"github.com/edaniels/golog"
+	ffmpeg "github.com/u2takey/ffmpeg-go"
+	"go.viam.com/test"
+	viamutils "go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+)
+
+func TestRTSPCamera(t *testing.T) {
+	logger := golog.NewTestLogger(t)
+	// set up the rtsp simple server running on rtsp://localhost:8598/mystream
+	outputURL := "rtsp://127.0.0.1:8598/mystream"
+	configLoc := []string{artifact.MustPath("components/camera/rtsp/rtsp-simple-server.yml")}
+	s, ok := server.New(configLoc)
+	test.That(t, ok, test.ShouldBeTrue)
+	defer s.Close()
+	// run the test mp4 in a loop through ffmpeg and hand it to the server
+	cancelCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	viamutils.PanicCapturingGo(func() {
+		testMP4 := artifact.MustPath("components/camera/rtsp/earth_480_mjpeg.avi")
+		ffmpegStream := ffmpeg.Input(testMP4, ffmpeg.KwArgs{"re": "", "stream_loop": -1})
+		ffmpegStream = ffmpegStream.Output(outputURL, ffmpeg.KwArgs{"vcodec": "mjpeg", "huffman": "0", "f": "rtsp", "rtsp_transport": "tcp"})
+		ffmpegStream.Context = cancelCtx
+		cmd := ffmpegStream.OverWriteOutput().Compile()
+		if err := cmd.Run(); err != nil {
+			return
+		}
+	})
+	time.Sleep(3 * time.Second) // wait so ffmpeg can publish to rtsp server
```

Comment:
Can we remove the sleep and change to signaling on a channel, to avoid flaky failures?

Reply:
I feel similar about this. More important though, I'd like to know what happens in a real use case when the rtsp stream isn't fully started when the camera component starts? If it fails, does it just rely on the component being "restarted" by the robot's config parser until it finally connects? Not saying that's unacceptable (if it works), but don't want to see robots fail if the startup sequencing means the rtsp server isn't up as quickly as needed.

Reply:
Essentially what happens is you get a 404 error when trying to connect the client to the server, and NewRTSPWebcam fails to be created. So yea, it relies on the component being restarted until the server is visible. 

I will check what happens if the RTSP server is turned off in the middle of running...

Reply:
OK - turns out that if the server is turned off and restarted in the middle of running, the stream stops and never begins again. So will add a reconnect-to-server function

Reply:
removed the sleep, and also added reconnection logic to the struct